pkg/local_object_storage/metabase/graveyard.go

```diff
@@ -261,6 +262,52 @@ func (db *DB) DropGraves(tss []TombstonedObject) error {
 	})
 }
 
+// DropExpiredTSMarks run through the graveyard and drops tombstone marks with
+// tombstones whose expiration is _less_ than provided epoch.
+// Returns number of marks dropped.
+func (db *DB) DropExpiredTSMarks(epoch uint64) (int, error) {
+	db.modeMtx.RLock()
+	defer db.modeMtx.RUnlock()
+
+	if db.mode.NoMetabase() {
+		return -1, ErrDegradedMode
+	} else if db.mode.ReadOnly() {
+		return -1, ErrReadOnlyMode
+	}
+
+	var counter int
+
+	err := db.boltDB.Update(func(tx *bbolt.Tx) error {
+		bkt := tx.Bucket(graveyardBucketName)
```

Comment:
what `if bkt == nil`?

Reply:
also not expected on an inited metabase: https://github.com/nspcc-dev/neofs-node/blob/b135a0efe9c6e38ce69f53f9dd1c191d62f218bd/pkg/local_object_storage/metabase/control.go#L104

Reply:
clear panic

Reply:
this operation is a process of `Inhume` and none of the methods in `inhume.go` does this check and panic. this is important to me cause it is literally "put" and then "get" with checking if the result is not nil, not sure bbolt code should be like this; and if we want this to be paniced with a desc, we have to add a desc in many more places

Reply:
as i said in https://github.com/nspcc-dev/neofs-node/pull/2937#discussion_r1771277638, we need to dedicate a new issue for this then. Could u pls create it?

Reply:
can you describe the problem so I can create it? as i have already said, panic is not expected _at all_. data is versioned and/or inited explicitly, any changed database data should be considered as a strange and not-possible-to-handle situation. panic with trace or panic with trace and two more words almost don't differ to me, dev will have to understand the whole situation anyway, and user will understand nothing. if values from bbolt should be expected corrupted we have to changed _many many_ places. considering all the said above, what do you want to be attached to a new issue?

Reply:
This can only happen with corrupted DB. Corrupted meta DB -> resync meta please. But making it a proper error in all cases won't be easy.